test/e2e/federated-service.go

```diff
@@ -168,8 +170,25 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 				}
 
 				createBackendPodsOrFail(clusters, nsName, FederatedServicePodName)
+
 				service = createServiceOrFail(f.FederationClientset_1_5, nsName, FederatedServiceName)
-				waitForServiceShardsOrFail(nsName, service, clusters)
+				obj, err := conversion.NewCloner().DeepCopy(service)
+				// Cloning shouldn't fail. On the off-chance it does, we should
+				// shallow copy service to serviceShard before failing.
+				if err != nil {
```

Comment:
Just fail the test. Not working cloner is something that should not go unnoticed.